lib/Macros/Sources/SwiftMacros/TaskLocalMacro.swift

```diff
@@ -0,0 +1,204 @@
+        syntax: declaration,
+        message: "'@TaskLocal' property must have default value, or be optional", id: .mustBeVar)
+    }
+
+    // If the property is global, do not prefix the synthesised decl with 'static'
+    let isGlobal = context.lexicalContext.isEmpty
+    let staticKeyword: String
+    if isGlobal {
+      staticKeyword = ""
+    } else {
+      staticKeyword = "static "
+    }
+
+    return [
+      """
+      \(raw: staticKeyword)let $\(name)\(raw: explicitType) = TaskLocal(wrappedValue: \(raw: initialValue))
+      """
+    ]
+  }
+}
+
+extension TaskLocalMacro: AccessorMacro {
+  public static func expansion(
+    of node: AttributeSyntax,
+    providingAccessorsOf declaration: some DeclSyntaxProtocol,
+    in context: some MacroExpansionContext
+  ) throws -> [AccessorDeclSyntax] {
+    // We very specifically have to fail and diagnose in the accessor macro,
+    // rather than in the peer macro, since returning [] from the accessor
+    // macro adds another series of errors about it missing to emit a decl.
+    guard let varDecl = try requireVar(declaration) else {
+      return []
+    }
+    try requireStaticContext(varDecl, in: context)
+
+    guard let firstBinding = varDecl.bindings.first else {
+      return [] // TODO: make error
+    }
+
+    guard let name = firstBinding.pattern.as(IdentifierPatternSyntax.self)?.identifier else {
+      return [] // TODO: make error
+    }
+
+    return ["get { $\(name).get() }"]
+  }
+}
+
+@discardableResult
+private func requireVar(_ decl: some DeclSyntaxProtocol,
+                        diagnose: Bool = true) throws -> VariableDeclSyntax? {
+  if let varDecl = decl.as(VariableDeclSyntax.self) {
+    return varDecl
+  }
+  if diagnose {
+    throw DiagnosticsError(
+      syntax: decl,
+      message: "'@TaskLocal' can only be applied to properties", id: .mustBeVar)
+  }
+
+  return nil
+}
+
+@discardableResult
+private func requireStaticContext(_ decl: VariableDeclSyntax,
+                                  in context: some MacroExpansionContext,
+                                  diagnose: Bool = true) throws -> Bool {
+  let isStatic = decl.modifiers.contains { modifier in
+    modifier.name.text == "\(Keyword.static)"
+  }
+
+  if isStatic {
+    return true
+  }
+
+  let isGlobal = context.lexicalContext.isEmpty
+  if isGlobal {
+    return true
+  }
+
+  if diagnose {
+    throw DiagnosticsError(
+      syntax: decl,
+      message: "'@TaskLocal' can only be applied to 'static' property", id: .mustBeStatic)
+  }
+
+  return false
+}
+
+extension TypeSyntax {
+  // This isn't great since we can't handle type aliases since the macro
+  // has no type information, but at least for the common case for Optional<T>
+  // and T? we can detect the optional.
+  fileprivate var isOptional: Bool {
+    let strRepr = "\(self)"
+    return strRepr.last == "?" ||
+      strRepr.starts(with: "Optional<") ||
+      strRepr.starts(with: "Swift.Optional<")
+  }
```

Comment:
This will fail as soon as there is whitespace involved, like `Optional < Int >`. What I would do instead is something like the following that inspects the tree structure

```swift
switch self.as(TypeSyntaxEnum.self) {
case .optionalType: 
  return true
case .identifierType(let identifierType): 
  return identifierType.name.text == "Optional"
case .memberType(let memberType): 
  guard let baseIdentifier = memberType.baseType.as(IdentifierTypeSyntax.self), baseIdentifier.name.text == "Swift" else {
    return false
  }
  return memberType.name.text == "Optional"
default: 
  return false
}
```

Reply:
Ah, whitespace -- thank you!

Reply:
Thanks a lot, this is much better